libraries/stdlib/genosl/mx_image_float.osl

```diff
@@ -1,8 +1,8 @@
 #include "$fileTransformUv"
 
-void mx_image_float(string file, string layer, float default_value, vector2 texcoord, string uaddressmode, string vaddressmode, string filtertype, string framerange, int frameoffset, string frameendaction, output float out)
+void mx_image_float(FILETEXTURE file, string layer, float default_value, vector2 texcoord, string uaddressmode, string vaddressmode, string filtertype, string framerange, int frameoffset, string frameendaction, output float out)
 {
-    if (file == "" ||
+    if (file.filename == "" ||
```

Comment:
[Newbie comment] That seems to be a major bug to request a texture with an empty file name.

Reply:
This is the current defined behaviour. i.e. If no texture then default value should be used.
One example is that this allows a single graph with an image node connected to a downstream input and either a texture or color can be used without rewiring the graph or requiring both a mapped and unmapped input.